### PR TITLE
Implemented Point, Shape and Triangle classes

### PR DESCRIPTION
**Pull request by nikitin-i**

### CODE REVIEW

File: src/Point.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+export class Point {
+    readonly initX: number = 0;
+    readonly initY: number = 0;
+    private _x: number;
+    private _y: number;
+
+    constructor();
+    constructor(x: number, y: number);
+    constructor(x?: number, y?: number) {
```
Comment (Hundd) [inline src/Point.ts:9]: You can use default values
```suggestion
    constructor(x?: number = 0, y?: number = 0) {
```

File: src/Point.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    distance();
+    distance(newX: number, newY: number);
+    distance(other: Point);
+    distance(newX?: any, newY?: any): number {
+        if (typeof newX === 'undefined' ) {
+            return Math.sqrt((this._x - this.initX) ** 2 + (this._y - this.initY) ** 2);
+        } else if (typeof newX === 'number' && typeof newY === 'number') {
+            return Math.sqrt((newX - this._x) ** 2 + (newY - this._y) ** 2);
+        } else if (typeof newX === 'object') {
+            return Math.sqrt((newX['_x'] - this._x) ** 2 + (newX['_y'] - this._y) ** 2);
```
Comment (Hundd) [inline src/Point.ts:27]: you can specify getter for `x` and `y` so that you don't rely on realization

File: src/Shape.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        this.color = userColor || 'green';
+        this.filled = typeof userFilled !== 'undefined' ? userFilled : true;
+
+        if (userPoints.length < 3) {
+            throw new Error('You have to pass three points!');
+        }
+    }
+
+    toString(): string {
+        const points: string = this.points.map(point => [`(${point['_x']}, ${point['_y']})`]).join(', ');
```
Comment (Hundd) [inline src/Shape.ts:21]: you can use toString method from Point class here

File: src/Shape.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+        return `A Shape with color of ${this.color} and ${this.filled ? 'filled' : 'not filled'}. Points: ${points}.`;
+    };
+
+    getPerimeter(): number {
+        const sideFirst: number = Math.round(this.points[0].distance(this.points[1]['_x'], this.points[1]['_y']) * 10) / 10;
```
Comment (Hundd) [inline src/Shape.ts:27]: please use methods defined in Point class for accepting another point. 
```
this.point[0].distanance(this.point[1])
```

File: src/Shape.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        const sideSecond: number = Math.round(this.points[1].distance(this.points[2]['_x'], this.points[2]['_y']) * 10) / 10;
```
Comment (Hundd) [inline src/Shape.ts:28]: if types can be calculated in run type there is no need specify them every time
const sideSecond = ...

File: src/Shape.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        const sideThird: number = Math.round(this.points[2].distance(this.points[0]['_x'], this.points[0]['_y']) * 10) / 10;
+
+        return sideFirst + sideSecond + sideThird;
```
Comment (Hundd) [inline src/Shape.ts:31]: Method should work with any number of Points

File: src/Triangle.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    getType(): string {
+        const sideFirst: number = Math.round(this.points[0].distance(this.points[1]['_x'], this.points[1]['_y']) * 10) / 10;
+        const sideSecond: number = Math.round(this.points[1].distance(this.points[2]['_x'], this.points[2]['_y']) * 10) / 10;
+        const sideThird: number = Math.round(this.points[2].distance(this.points[0]['_x'], this.points[0]['_y']) * 10) / 10;
+
+        const isEquilateral: string = (sideFirst === sideSecond && sideFirst === sideThird && sideSecond === sideThird) && 'equilateral triangle';
+        const isIsosceles: string = (sideFirst === sideSecond || sideFirst === sideThird || sideSecond === sideThird) && 'isosceles triangle';
+        const isScalene: string = (sideFirst !== sideSecond && sideFirst !== sideThird && sideSecond !== sideThird) && 'scalene triangle';
```
Comment (Hundd) [inline src/Triangle.ts:32]: `is` prefix generally have Boolean variables

Comment (Hundd) [reply src/Triangle.ts:32]: you can simplify this a bit by using `Set`
```js
const sizes = new Set([sideFirst, sideSecond, sideThird]).size
return [,'equilateral triangle', 'isosceles triangle', 'scalene triangle'][sizes]
```

**Review by Hundd — CHANGES_REQUESTED** [review]

**Comment by nikitin-i** [comment]
Closed

**Pull request closed without merging**